src/Controllers/DatabaseController.js

```diff
@@ -1751,18 +1752,28 @@ class DatabaseController {
       logger.warn('Unable to ensure uniqueness for role name: ', error);
       throw error;
     });
+
+    await this.adapter
+      .ensureUniqueness('_Idempotency', requiredIdempotencyFields, ['reqId'])
+      .catch(error => {
+        logger.warn('Unable to ensure uniqueness for idempotency request ID: ', error);
+        throw error;
+      });
+
     if (this.adapter instanceof MongoStorageAdapter) {
       await this.adapter
-        .ensureUniqueness('_Idempotency', requiredIdempotencyFields, ['reqId'])
+        .ensureIndex('_Idempotency', requiredIdempotencyFields, ['expire'], 'ttl', false, {
+          ttl: 0,
+        })
         .catch(error => {
-          logger.warn('Unable to ensure uniqueness for idempotency request ID: ', error);
+          logger.warn('Unable to create TTL index for idempotency expire date: ', error);
           throw error;
         });
-
+    } else if (this.adapter instanceof PostgresStorageAdapter) {
+      const options = this.idempotencyOptions;
+      options.setIdempotencyFunction = true;
```

Comment:
If I see the diff correctly, the code block for postgres seems to be almost a duplicate of the one for mongo, expect for `options.setIdempotencyFunction = true;`. Could you dedupe this?

Reply:
I tried before, but the mongo adapter began failing for some reason I couldn't figure out. I'll try again to see if the problem I had persists, or if I can implement it a different way

Reply:
Lets see if this change passes on mongo and let me know if it looks good

Reply:
Looks like it's good to go...